### DISCUSSION TITLE
Add ability to use include and exclude inside DTO construction functions?

### DISCUSSION THREAD

**Discussion by looqey** [spatie/laravel-data#465] [Ideas]
Hello,

I have an idea that could potentially improve the experience of using this library. Let's consider a data class called UserData, which extends the Data class:

```php
class UserData extends Data {
    public function __construct(
          public int $id,
          public string $name,
          public RoleData|Lazy $role
     ) {}
   // ...
}
```
Now, let's say we have some other data classes that use the UserData class. In these classes, sometimes we need to include the RoleData property, and sometimes we don't. Currently, when we need to include it, we have to use include('user.role') when creating the DTO object of the class. Here's an example:

```php
class FacilityData extends Data {
    public function __construct(
        public int $id,
        public LocationData $location,
        public UserData $user
    ) {}

    public static function fromModel(Facility $facility): self {
         return new self(
               $facility->id,
               $facility->name,
               UserData::fromModel($facility->owner)->include('role') // this include will be ignored...
        );
    }
}

// Client code
$data = FacilityData::fromModel($facility)->include('user.role'); // ..so we have to apply includes to the root data object during creation
```
In the fromModel construction method of FacilityData, if we try to include the $role property, it will be ignored. However, having this ability would be useful in many cases, especially when we have a tree of data objects that need to be included in a sequence, requiring the use of include() with long paths.

**Comment by rubenvanassche** [comment]
This should be normally fixed with data v4, the inclusion system is completely rewritten over there to allow these kinds of constructions. Expect the upgrade later this year!
